### PR TITLE
test: Addition of test cases for the Data Preprocessor code

### PR DESCRIPTION
**Pull request by Abhishek-TAMU**
### Description of the change

Addition of unit test cases for function: [load_dataset](https://github.com/dushyantbehl/fms-hf-tuning/blob/c078d071beeadef718dd4d645655658ae5391e9d/tuning/data/data_processors.py#L75C9-L75C21) and function [_process_dataconfig_file](https://github.com/dushyantbehl/fms-hf-tuning/blob/c078d071beeadef718dd4d645655658ae5391e9d/tuning/data/setup_dataprocessor.py#L60C5-L60C29)

### Related issue number

https://github.ibm.com/ai-foundation/watson-fm-stack-tracker/issues/1428

### How to verify the PR

Running test cases

### Was the PR tested

- [ ] I have added >=1 unit test(s) for every new method I have added.
- [ ] I have ensured all unit tests pass

### CODE REVIEW

File: tests/data/test_data_preprocessing_utils.py [hunk 1]
```diff
@@ -18,12 +23,30 @@
...
+PREDEFINED_DATA_CONFIGS = os.path.join(BASE_DIR, "examples", "predefined_data_configs")
+APPLY_CUSTOM_TEMPLATE_YAML = os.path.join(
+    PREDEFINED_DATA_CONFIGS, "apply_custom_template.yaml"
+)
+PRETOKENIZE_JSON_DATA_YAML = os.path.join(
+    PREDEFINED_DATA_CONFIGS, "pretokenized_json_data.yaml"
+)
+TOKENIZE_AND_INSTRUCTION_MASKING_YAML = os.path.join(
+    PREDEFINED_DATA_CONFIGS, "tokenize_and_instruction_masking.yaml"
+)
```
Comment (willmj) [inline tests/data/test_data_preprocessing_utils.py]: Should this go in an \_\_init\_\_.py file like the one for [tests/testdata](https://github.com/dushyantbehl/fms-hf-tuning/blob/c078d071beeadef718dd4d645655658ae5391e9d/tests/testdata/__init__.py)?

**Review by willmj — COMMENTED** [review]
Test cases look good, one thing before approval:

File: tests/data/test_data_preprocessing_utils.py (hunk 1 above)
Comment (Abhishek-TAMU) [reply tests/data/test_data_preprocessing_utils.py]: Yea make sense. Thanks Will. Pushed the refactored code.

**Review by willmj — APPROVED** [review]
LGTM! Thanks Abhishek

**Pull request closed without merging**